contracts/v3/converters/General2Converter.sol

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/math/SafeMath.sol';
+import '@openzeppelin/contracts/token/ERC20/ERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+
+import '../interfaces/IConverter.sol';
+import '../interfaces/IManager.sol';
+import '../interfaces/ICurve2Pool.sol';
+
+/**
+ * @title General2Converter
+ */
+contract General2Converter is IConverter {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    IManager public immutable override manager;
+    ICurve2Pool public immutable stableSwap2Pool;
+    IERC20 public immutable tokenCRV; // 2Crv
+
+    IERC20[2] public tokens;
+
+    mapping(address => int128) internal indices;
+
+    /**
+     * @param _tokens A list of the address of the token in the Curve pool
+     * @param _tokenCRV The address of the CRV token
+     * @param _stableSwap2Pool The address of 2Pool
+     * @param _manager The address of the Vault Manager
+     */
+    constructor(
+        IERC20[] memory _tokens,
+        IERC20 _tokenCRV,
+        ICurve2Pool _stableSwap2Pool,
+        IManager _manager
+    ) public {
+        require(_tokens.length == 2, 'Token count must be 2');
+
+        tokenCRV = _tokenCRV;
+        stableSwap2Pool = _stableSwap2Pool;
+        manager = _manager;
+
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            tokens[i] = _tokens[i];
```

Comment:
A (small) risk is that a token is (accidentally) supplied twice, which would lead to strange results. This is easily checked by adding the following:
   require(indices[address(_tokens[i])] == 0,"Double"); // xtra